index.js

```diff
@@ -1,96 +1,565 @@
-    ImportSpecifier: [maybeNode("name"), node("binding")],
-    LabeledStatement: [node("label"), node("body")],
-    LiteralBooleanExpression: [primitive("value")],
-    LiteralInfinityExpression: [],
-    LiteralNullExpression: [],
-    LiteralNumericExpression: [primitive("value")],
-    LiteralRegExpExpression: [primitive("pattern"), primitive("flags")],
-    LiteralStringExpression: [primitive("value")],
-    Method: [primitive("isGenerator"), node("name"), listNode("parameters"), maybeNode("restParameter"), node("body")],
-    Module: [listNode("items")],
-    NewExpression: [node("callee"), listNode("arguments")],
-    NewTargetExpression: [],
-    ObjectBinding: [listNode("properties")],
-    ObjectExpression: [listNode("properties")],
-    PostfixExpression: [node("operand"), primitive("operator")],
-    PrefixExpression: [primitive("operator"), node("operand")],
-    ReturnStatement: [maybeNode("expression")],
-    Script: [node("body")],
-    Setter: [node("name"), node("parameter"), node("body")],
-    ShorthandProperty: [node("name")],
-    SpreadElement: [node("expression")],
-    StaticMemberExpression: [node("object"), primitive("property")],
-    StaticPropertyName: [primitive("value")],
-    Super: [],
-    SwitchCase: [node("test"), listNode("consequent")],
-    SwitchDefault: [listNode("consequent")],
-    SwitchStatement: [node("discriminant"), listNode("cases")],
-    SwitchStatementWithDefault: [node("discriminant"), listNode("preDefaultCases"), node("defaultCase"), listNode("postDefaultCases")],
-    TemplateElement: [primitive("rawValue")],
-    TemplateExpression: [maybeNode("tag"), listNode("elements")],
-    ThisExpression: [],
-    ThrowStatement: [node("expression")],
-    TryCatchStatement: [node("body"), node("catchClause")],
-    TryFinallyStatement: [node("body"), maybeNode("catchClause"), node("finalizer")],
-    VariableDeclaration: [primitive("kind"), listNode("declarators")],
-    VariableDeclarationStatement: [node("declaration")],
-    VariableDeclarator: [node("binding"), maybeNode("init")],
-    WhileStatement: [node("test"), node("body")],
-    WithStatement: [node("object"), node("body")],
-    YieldExpression: [maybeNode("expression")],
-    YieldGeneratorExpression: [node("expression")],
+/**
+ * Copyright 2015 Shape Security, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+exports.genSpec = function (Maybe, List, Union){
```

Comment:
I still don't see why these need to be passed in instead of us defining our own ADTs.